adapters/connectad/connectad.go

```diff
@@ -0,0 +1,209 @@
+package connectad
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type ConnectAdAdapter struct {
+	endpoint string
+}
+
+type connectadImpExt struct {
+	ConnectAd openrtb_ext.ExtImpConnectAd `json:"connectad"`
+}
+
+func NewConnectAdBidder(endpoint string) *ConnectAdAdapter {
+	return &ConnectAdAdapter{
+		endpoint: endpoint,
+	}
+}
+
+func (a *ConnectAdAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	var errs []error
+
+	if errs := preprocess(request); len(errs) > 0 {
+		return nil, append(errs, &errortypes.BadInput{
```

Comment:
The error message is being appended to the list of errors, making the including of the `errs` as a string redundant. Please consider passing the errors directly through or removing the `errs` part of the general message if you still want to provide clear direction for where the errors are coming from.